src/shared/components/setup-locker/SetupLocker.js

```diff
@@ -1,52 +1,63 @@
 import React, { Component } from 'react';
+import { FlowModal, FlowModalStep, setAppElement } from '@nomios/web-uikit';
 import PropTypes from 'prop-types';
 
-import SetupMasterLock from './steps/SetupMasterLock';
-import SetupIdleTimer from './steps/SetupIdleTimer';
+import SetPassphrase from './setup-locker-steps/set-passphrase/SetPassphrase';
+import SetTimeout from './setup-locker-steps/set-timeout/SetTimeout';
+
+setAppElement('#root');
+
+const LOCK_TYPE = 'passphrase';
 
 class SetupLocker extends Component {
+    constructor(props) {
+        super(props);
+        const locker = props.locker;
+
+        this.analyzePasswordQuality = locker.getLock(LOCK_TYPE).validate.bind(locker.getLock(LOCK_TYPE));
+        this.enablePassword = locker.getLock(LOCK_TYPE).enable.bind(locker.getLock(LOCK_TYPE));
+        this.setMaxTime = locker.getIdleTimer().setMaxTime.bind(locker.getIdleTimer());
+    }
+
     state = {
-        step: 1,
+        stepId: 'passphrase',
+        data: {},
     };
 
     render() {
+        const { stepId } = this.state;
+        const { isOpen } = this.props;
+
         return (
-            <div>
-                <h1>Setup Locker</h1>
-                { this.renderStep() }
-            </div>
+            <FlowModal open={ isOpen } variant="simple" step={ stepId } showClose={ false }>
```

Comment:
Other Modal props should be supported and spread here. For instance, we will need to attach `onExited` to open the create/import flow afterwards.